opendbc/car/toyota/carcontroller.py

```diff
@@ -59,6 +64,18 @@ def update(self, CC, CS, now_nanos):
     # *** control msgs ***
     can_sends = []
 
+    # *** handle secoc reset counter increase ***
+    if self.CP.flags & ToyotaFlags.SECOC.value:
+      if CS.secoc_synchronization['RESET_CNT'] != self.secoc_prev_reset_counter:
+        self.secoc_lka_message_counter = 0
+        self.secoc_lta_message_counter = 0
+        self.secoc_prev_reset_counter = CS.secoc_synchronization['RESET_CNT']
+
+        # Verify mac of SecOC synchronization message to ensure we have the right key
+        # TODO: move this to tests rather than runtime
+        # expected_mac = build_sync_mac(self.secoc_key, int(CS.secoc_synchronization['TRIP_CNT']), int(CS.secoc_synchronization['RESET_CNT']))
+        # assert(int(CS.secoc_synchronization['AUTHENTICATOR']) == expected_mac)
```

Comment:
If you want to show an alert for having the wrong SecOC key this would be great place to detect that.

Reply:
Gave it some thought, but at the time I was trying to get the minimum viable port running and we can't use cloudlog here anymore. What we'd have to do is add a CarState counter, and then add some logic in openpilot `card` to surface an alert after some number of violations.

Right now the cars go to LKA fault on a missing key, but we should be showing the dashcam no-security-key startup alert on top of it. For an actually wrong key, that could use detection eventually.

Compromise for now: I'll add the CarState counter without the event logic. Quick, easy, it'll help us troubleshoot if the problem ever comes pu, and we'll get some data from the field on how to set alerting thresholds if we decide it's needed.

Reply:
We forward logs from carlog to cloudlog: https://github.com/commaai/opendbc/blob/739117d561b265de5caf843c3f117490a9de57e6/opendbc/car/__init__.py#L14-L17

Reply:
sweet, will implement